test/unit/mock/readable-storage-logic.spec.ts

```diff
@@ -0,0 +1,175 @@
+import { MockContract, MockContractFactory, smock } from '@src';
+import { convertStructToPojo } from '@src/utils';
+import { ADDRESS_EXAMPLE, BYTES32_EXAMPLE, BYTES_EXAMPLE } from '@test-utils';
+import { StorageGetter, StorageGetter__factory } from '@typechained';
+import { expect } from 'chai';
+import { BigNumber, utils } from 'ethers';
+
+describe('Mock: Readable storage logic', () => {
+  let storageGetterFactory: MockContractFactory<StorageGetter__factory>;
+  let mock: MockContract<StorageGetter>;
+
+  before(async () => {
+    storageGetterFactory = await smock.mock('StorageGetter');
+  });
+
+  beforeEach(async () => {
+    mock = await storageGetterFactory.deploy(1);
+  });
+
+  describe('getVariable', () => {
+    it('should be able to get a uint256', async () => {
+      const value = utils.parseUnits('123');
+      await mock.setVariable('_uint256', value);
+      expect(await mock.getUint256()).to.equal(value);
+
+      const getValue = await mock.getVariable('_uint256');
+      const expectedValue = await mock.getUint256();
+      expect(getValue).to.equal(expectedValue);
+    });
+
+    it('should be able to get a uint16 in a packed slot', async () => {
+      const value = BigNumber.from('1');
+      const value2 = BigNumber.from('2');
+      await mock.setPackedUintA(value);
+      await mock.setPackedUintB(value2);
+      expect(await mock.getPackedUintA()).to.equal(value);
+      expect(await mock.getPackedUintB()).to.equal(value2);
+
+      const getValue = await mock.getVariable('_packedUintB');
+      const expectedValue = await mock.getPackedUintB();
+      expect(getValue).to.equal(expectedValue);
+    });
+
+    it('should be able to get a int56', async () => {
+      const value = 1;
+      await mock.setVariable('_int56', value);
+      expect(await mock.getInt56()).to.equal(value);
+
+      const getValue = await mock.getVariable('_int56');
+      const expectedValue = await mock.getInt56();
+      expect(getValue.toString()).to.equal(expectedValue.toString());
+    });
+
+    // it('should be able to get a int256', async () => {
+    //   const value = utils.parseUnits('-1');
+    //   await mock.setVariable('_int256', value);
+
+    //   const getValue = await mock.getVariable('_int256');
+    //   const expectedValue = await mock.getInt256();
+    //   console.log("getValue: ", getValue);
+    //   expect(getValue).to.equal(expectedValue.toString());
+    // });
+
+    it('should be able to get an address', async () => {
+      await mock.setVariable('_address', ADDRESS_EXAMPLE);
+      expect(await mock.getAddress()).to.equal(ADDRESS_EXAMPLE);
+
+      const getValue = await mock.getVariable('_address');
+      const expectedValue = await mock.getAddress();
+      expect(getValue).to.equal(expectedValue);
+    });
```

Comment:
again, i have the feeling that both expectations are redundant
we're not testing `setVariable` here, so i guess you can trust that `expect(await mock.getAddress()).to.equal(ADDRESS_EXAMPLE)` will be ok